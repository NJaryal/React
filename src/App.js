import React from 'react';
import MainContainer from 'src/MainContainer';
import NotFound from 'src/NotFound/NotFound';
import ErrorBoundary from 'Components/common/ErrorBoundary';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import store from 'src/redux/store';
import history from "src/redux/history";

const App = () => (
  <Provider store={store}>
    <Router history={history}>
      <ErrorBoundary>
        <Route path="/" exact component={MainContainer} />
        <Route path="/film/:id" component={MainContainer} />
        <Route component={NotFound} />
      </ErrorBoundary>
    </Router>
  </Provider>
);

export default App;
