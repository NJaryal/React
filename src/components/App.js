import React from 'react';
import MainContainer from './MainContainer/MainContainer';
import ErrorBoundary from './common/ErrorBoundary';

const App = () => (
  <ErrorBoundary>
    <MainContainer />
  </ErrorBoundary>
);

export default App;
