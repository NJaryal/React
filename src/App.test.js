import React from 'react';
import { shallow } from 'enzyme';
import 'babel-polyfill';

import ReactDOM from 'react-dom';
import App from './App';

const setup = () => {
  const component = shallow(<App />);
  return component;
};

describe('App Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });

  it('should render Component', () => {
    expect(component).toMatchSnapshot();
  });
});

