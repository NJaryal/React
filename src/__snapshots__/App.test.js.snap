// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render Component 1`] = `<Routes />`;
